### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by tantaman** [vlcn-io/cr-sqlite#347] [Announcements]
# Major Focus Areas

- More CRDT types https://github.com/vlcn-io/cr-sqlite/issues/322
- Improved reactivity https://github.com/vlcn-io/cr-sqlite/discussions/309
  -  Fully generic reactivity that incrementally updates queries will be a paid feature. Free for indie devs, however.
- TypedSQL & TreeSQL https://github.com/vlcn-io/typed-sql & https://github.com/tantaman/TreeSQL
- Supporting query based sync of strongly consistent data through something like https://twitter.com/tantaman/status/1699464628072296706
- Higher level networking primitives https://github.com/vlcn-io/cr-sqlite/issues/38

# Smaller Initiatives

- Improve [Turso](https://turso.tech/) and [LiteFS](https://fly.io/docs/litefs/) support
- Add query based sync examples & tutorials
- Reducing CRDT metadata size
- Improve the vite starter to be "choose your own adventure"
  - react or vanilla js
  - typed sql or no typed sql
  - ts or no ts
  - clerk auth or no auth
  - websocket sync or rest sync
- Complete the virtual table CRR creation syntax - https://github.com/vlcn-io/cr-sqlite/pull/324
- Whole row sync to remove the constraint that all columns have default values

# Future Big Bets

- Postgres integration
- Sync as a service

**Comment by tantaman** [comment]
The `JS` and reactivity roadmap items should be removed from the `cr-sqlite` roadmap as they are part of different components. Looks like its time to have some broader place to roll individual roadmaps into.

**Comment by MuriloSchaefer** [comment]
Is this project still maintained?
